scipy/special/tests/test_cdft_asymptotic.py

```diff
@@ -0,0 +1,49 @@
+# gh-14777 regression tests
+# Test stdtr and stdtrit with infinite df and large values of df
+
+import numpy as np
+from numpy.testing import assert_allclose, assert_equal
+from scipy.special import stdtr, stdtrit, ndtr, ndtri
+
+
+def test_stdtr_vs_R_large_df():
+    df = [1e10, 1e12, 1e120, np.inf]
+    t = 1.
+    res = stdtr(df, t)
+    # R Code:
+    #   options(digits=20)
+    #   pt(1., c(1e10, 1e12, 1e120, Inf))
+    res_R = [0.84134474605644460343,
+             0.84134474606842180044,
+             0.84134474606854281475,
+             0.84134474606854292578]
+    assert_allclose(res, res_R, rtol=1e-15)
```

Comment:
```suggestion
    assert_allclose(res, res_R, rtol=2e-15)
```
As Ralf mentioned, on 32-bit Ubuntu CI:
```
___________________________ test_stdtr_vs_R_large_df ___________________________
[gw1] linux -- Python 3.8.0 /usr/bin/python3.8
scipy/special/tests/test_cdft_asymptotic.py:20: in test_stdtr_vs_R_large_df
    assert_allclose(res, res_R, rtol=1e-15)
E   AssertionError: 
E   Not equal to tolerance rtol=1e-15, atol=0
E   
E   Mismatch: 25%
E   Max absolute difference: 8.8817842e-16
E   Max relative difference: 1.05566526e-15
E    x: array([0.841345, 0.841345, 0.841345, 0.841345])
E    y: array([0.841345, 0.841345, 0.841345, 0.841345])
        df         = [10000000000.0, 1000000000000.0, 1e+120, inf]
        res        = array([0.84134475, 0.84134475, 0.84134475, 0.84134475])
        res_R      = [0.8413447460564446, 0.8413447460684218, 0.8413447460685428, 0.8413447460685429]
        t          = 1.0
- generated xml file: /scipy/build/testenv/lib/python3.8/site-packages/junit/test-results.xml -
=========================== short test summary info ============================
```

Reply:
Thanks!